## Exercitiul 1
(fizz-buzz test) Sa se scrie numerele de la 1 la n; pentru fiecare multiplu de 2 se va scrie in locul numarului 'Fizz', pentru multiplu al lui 5 se va scrie 'Buzz'; daca numarul este multiplu de 10 se va scrie in locul lui 'FizzBuzz'.

In [6]:
def FizzBuzz(n):
    for element in range(1, n+1):
        if element % 10 == 0:
            print('FizzBuzz')
        elif element % 2 == 0:
            print('Fizz')
        elif element % 5 == 0:
            print('Buzz')
        else:
            print(element)

In [7]:
FizzBuzz(10)

1
Fizz
3
Fizz
Buzz
Fizz
7
Fizz
9
FizzBuzz


## Exercitiul 2
Sa se calculeze scorul Scrabble al unui cuvant. Fiecare litera are asignata o anumita valoare (e.g. 'a' -> 3, 'f' ->7, 'z' -> 23; setati valori intr-un dictionar), iar scorul cuvantului se calculeaza ca suma scorurilor literelor componente. Literele mari si mici au aceeasi valoare.

In [44]:
dictionarScrabble = {
    'a' : 1, 
    'b' : 2,
    'c' : 3,
    'd' : 4,
    'e' : 5,
    'f' : 6,
    'g' : 7,
    'h' : 8,
    'i' : 9,
    'j' : 10,
    'k' : 11,
    'l' : 12,
    'm' : 13,
    'n' : 14,
    'o' : 15,
    'p' : 16,
    'q' : 17,
    'r' : 18,
    's' : 19,
    't' : 20,
    'u' : 21,
    'v' : 22,
    'w' : 23,
    'x' : 24,
    'y' : 25,
    'z' : 26,
}

In [45]:
def scrabble(cuvant):
    scor = 0
    cuvant = cuvant.lower()
    for litera in cuvant:
        scor += dictionarScrabble[litera]
    return scor

In [46]:
print(Scrabble('Alfabet'))

47


## Exercitiul 3
Sa se determine daca doua cuvinte sunt unul anagrama celuilalt; exemplu LIBIA -> ABILI. Considerati sortarea caracterelor unui cuvant.

In [17]:
def anagrama(cuvant1, cuvant2):
    cuvant1, cuvant2 = cuvant1.lower(), cuvant2.lower()
    cuvant1 = ''.join(sorted(cuvant1))
    cuvant2 = ''.join(sorted(cuvant2))
    if cuvant1 == cuvant2:
        print('Cuvintele sunt anagrame')
    else:
        print('Cuvintele nu sunt anagrame')

In [20]:
anagrama('Libia', 'abili')

Cuvintele sunt anagrame


## Exercitiul 4
Sa se verifice conjectura Collatz pentru numerele intre 1 si 1000.

In [21]:
def conjectura(numar):
    print(numar)
    while numar != 1:
        if numar % 2 == 0:
            numar = numar // 2
        else:
            numar = 3 * numar + 1
        print(numar)

In [25]:
for numar in range(1, 1001):
#     print('Numar ')
    conjectura(numar)
    print('*****************')

1
*****************
2
1
*****************
3
10
5
16
8
4
2
1
*****************
4
2
1
*****************
5
16
8
4
2
1
*****************
6
3
10
5
16
8
4
2
1
*****************
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
8
4
2
1
*****************
9
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
10
5
16
8
4
2
1
*****************
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
12
6
3
10
5
16
8
4
2
1
*****************
13
40
20
10
5
16
8
4
2
1
*****************
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
15
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
16
8
4
2
1
*****************
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
18
9
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
20
10
5
16
8
4
2
1
*****************
21
64
32
16
8
4
2
1
*****************
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*********

48
24
12
6
3
10
5
16
8
4
2
1
*****************
97
292
146
73
220
110
55
166
83
250
125
376
188
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
99
298
149
448
224
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
100
50
25
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
102
51
154
77
232
116
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
************

29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
156
78
39
118
59
178
89
268
134
67
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
157
472
236
118
59
178
89
268
134
67
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
158
79
238
119
358
179
538
269
808
404
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
159
478
239
718
359
1078
539
1618
809
2428
1214
607
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
6

23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
208
104
52
26
13
40
20
10
5
16
8
4
2
1
*****************
209
628
314
157
472
236
118
59
178
89
268
134
67
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
210
105
316
158
79
238
119
358
179
538
269
808
404
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
211
634
317
952
476
238
119
358
179
538
269
808
404
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
212
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
213
640
320
160
80
40
20
10
5
16
8
4
2
1
*****************
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732


325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
258
129
388
194
97
292
146
73
220
110
55
166
83
250
125
376
188
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
259
778
389
1168
584
292
146
73
220
110
55
166
83
250
125
376
188
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
8

16
8
4
2
1
*****************
311
934
467
1402
701
2104
1052
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
312
156
78
39
118
59
178
89
268
134
67
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
313
940
470
235
706
353
1060
530
265
796
398
199
598
299
898
449
1348
674
337
1012
506
253
760
380
190
95
286
143
430
215
646
323
970
485
1456
728
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976

152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
358
179
538
269
808
404
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
359
1078
539
1618
809
2428
1214
607
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
360
180
90
45
136
68
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
361
1084
542
271
814
407
1222
611
1834
917
2752
1376
688
344
172
86
43
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
362
181
544
272
136
68
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
363
1090
545
1636
818
409
1228
614
307
922
461
1384
692
346
173
520
260
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
3

74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
408
204
102
51
154
77
232
116
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
409
1228
614
307
922
461
1384
692
346
173
520
260
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
410
205
616
308
154
77
232
116
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
411
1234
617
1852
926
463
1390
695
2086
1043
3130
1565
4696
2348
1174
587
1762
881
2644
1322
661
1984
992
496
248
124
62
31
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*********

226
113
340
170
85
256
128
64
32
16
8
4
2
1
*****************
453
1360
680
340
170
85
256
128
64
32
16
8
4
2
1
*****************
454
227
682
341
1024
512
256
128
64
32
16
8
4
2
1
*****************
455
1366
683
2050
1025
3076
1538
769
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
456
228
114
57
172
86
43
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
457
1372
686
343
1030
515
1546
773
2320
1160
580
290
145
436
218
109
328
164
82
41
124
62
31
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5

244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
501
1504
752
376
188
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
503
1510
755
2266
1133
3400
1700
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308

44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
547
1642
821
2464
1232
616
308
154
77
232
116
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
548
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
549
1648
824
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
550
275
826
413
1240
620
3

152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
592
296
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
594
297
892
446
223
670
335
1006
503
1510
755
2266
1133
3400
1700
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
595
1786
893
2680
1340
670
335
1006
503
1510
755
2266
1133
3400
1700
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
20

86
43
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
642
321
964
482
241
724
362
181
544
272
136
68
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
643
1930
965
2896
1448
724
362
181
544
272
136
68
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
644
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
645
1936
968
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
273

32
16
8
4
2
1
*****************
683
2050
1025
3076
1538
769
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
684
342
171
514
257
772
386
193
580
290
145
436
218
109
328
164
82
41
124
62
31
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
685
2056
1028
514
257
772
386
193
580
290
145
436
218
109
328
164
82
41
124
62
31
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
31

5
16
8
4
2
1
*****************
724
362
181
544
272
136
68
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
725
2176
1088
544
272
136
68
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
726
363
1090
545
1636
818
409
1228
614
307
922
461
1384
692
346
173
520
260
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
727
2182
1091
3274
1637
4912
2456
1228
614
307
922
461
1384
692
346
173
520
260
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
728
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
729
2188
1094
547
1642
821
2464
123

310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
764
382
191
574
287
862
431
1294
647
1942
971
2914
1457
4372
2186
1093
3280
1640
820
410
205
616
308
154
77
232
116
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
765
2296
1148
574
287
862
431
1294
647
1942
971
2914
1457
4372
2186
1093
3280
1640
820
410
205
616
308
154
77
232
116
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
766
383
1150
575
1726
863
2590
1295
3886
1943
5830
2915
8746
4373
13120
6560
3280
1640
820
410
205
616
308
154
77
232
116
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
767
2302
1151
3454
1727
5182
2591
7774
3887
11662
5831

16
8
4
2
1
*****************
799
2398
1199
3598
1799
5398
2699
8098
4049
12148
6074
3037
9112
4556
2278
1139
3418
1709
5128
2564
1282
641
1924
962
481
1444
722
361
1084
542
271
814
407
1222
611
1834
917
2752
1376
688
344
172
86
43
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
800
400
200
100
50
25
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
801
2404
1202
601
1804
902
451
1354
677
2032
1016
508
254
127
382
191
574
287
862
431
1294
647
1942
971
2914
1457
4372
2186
1093
3280
1640
820
410
205
616
308
154
77
232
116
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
802
401
1204
602
301
904
452
226
113
340
170
85
256
128
64
32
16
8
4
2
1
*****************
803
2410
1205
3616
1808
904
452
226
113
340
170
85
256
128
64
32
16
8
4
2
1
*****************
804
402
201
604
302
151
454
227
682
341
1024
512
256
128
64
32
16
8
4
2
1
*****************
805
2416
1208
604
302
151
454
227
682


1889
5668
2834
1417
4252
2126
1063
3190
1595
4786
2393
7180
3590
1795
5386
2693
8080
4040
2020
1010
505
1516
758
379
1138
569
1708
854
427
1282
641
1924
962
481
1444
722
361
1084
542
271
814
407
1222
611
1834
917
2752
1376
688
344
172
86
43
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
840
420
210
105
316
158
79
238
119
358
179
538
269
808
404
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
841
2524
1262
631
1894
947
2842
1421
4264
2132
1066
533
1600
800
400
200
100
50
25
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
842
421
1264
632
316
158
79
238
119
358
179
538
269
808
404
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
843
2530
1265
3796
1898
949
2848
1424
712
356
178
89
268
134
67
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
844
422
211
63

92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
881
2644
1322
661
1984
992
496
248
124
62
31
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
882
441
1324
662
331
994
497
1492
746
373
1120
560
280
140
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
883
2650
1325
3976
1988
994
497
1492
746
373
1120
560
280
140
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
884
442
221
664
332
166
83
250
125
376
188
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502


26
13
40
20
10
5
16
8
4
2
1
*****************
918
459
1378
689
2068
1034
517
1552
776
388
194
97
292
146
73
220
110
55
166
83
250
125
376
188
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
919
2758
1379
4138
2069
6208
3104
1552
776
388
194
97
292
146
73
220
110
55
166
83
250
125
376
188
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308


452
226
113
340
170
85
256
128
64
32
16
8
4
2
1
*****************
952
476
238
119
358
179
538
269
808
404
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
953
2860
1430
715
2146
1073
3220
1610
805
2416
1208
604
302
151
454
227
682
341
1024
512
256
128
64
32
16
8
4
2
1
*****************
954
477
1432
716
358
179
538
269
808
404
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
955
2866
1433
4300
2150
1075
3226
1613
4840
2420
1210
605
1816
908
454
227
682
341
1024
512
256
128
64
32
16
8
4
2
1
*****************
956
478
239
718
359
1078
539
1618
809
2428
1214
607
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
957
2872
1436
718
359
1078
539
1618
809
2428
1214
607
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
12

1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
992
496
248
124
62
31
94
47
142
71
214
107
322
161
484
242
121
364
182
91
274
137
412
206
103
310
155
466
233
700
350
175
526
263
790
395
1186
593
1780
890
445
1336
668
334
167
502
251
754
377
1132
566
283
850
425
1276
638
319
958
479
1438
719
2158
1079
3238
1619
4858
2429
7288
3644
1822
911
2734
1367
4102
2051
6154
3077
9232
4616
2308
1154
577
1732
866
433
1300
650
325
976
488
244
122
61
184
92
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
*****************
993
2980
1490
745
2236
1118
559
1678
839
2518
1259
3778
1889
5668
2834
1417
4252
2126
1063
3190
1595
4786
2393
7180
3590
1795
5386
2693
8080
4040
2020
1010
505
1516
758
379
1138
569
1708
854
427
1282
641
1924
962
481
1444
722
361
1084
542
271
814
407
1222
611
1834
917
2752
1376
688
344
172
86
43
130
65
196
98
49
148
74
37
112
56
28
14
7
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
*****************
994
497
1492
746
373
1120
560


## Exercitiul 5
Sa se determine daca doua liste sunt una permutarea circulara a celeilalte. Exemplu: [1, 2, 3, 4] si [3, 4, 1, 2] sunt in aceasta situatie, dar nu si [1, 2, 3, 4] cu [3, 4, 2, 1].

In [27]:
# def permutare(lista1, lista2):
#     for element in lista1:
        

## Exercitiul 6
Plecand de la o lista cu numere naturale, sa se creeze numarul obtinut prin concatenarea cifrelor existente. Exemplu: [11, 3, 24, 0, 993] -> 113240993.

In [29]:
def creareNumar(lista):
    numar = ''
    for element in lista:
        numar += str(element)
    return int(numar)

In [31]:
numar = creareNumar([11,3,24,0,993])
print(numar)

113240993


## Exercitiul 7
Sa se creeze o functie care preia un numar n si returneaza un alt numar pe baza cifrelor lui n, astfel: se calculeaza numarul cifrelor pare din n (fie el si 0), numarul de cifre impare (poate fi si 0), suma lor si se formeaza numarul din acestea 2; daca exista zero nesemnificativ, acesta se va ignora. Numarul obtinut se va supune aceleiasi transformari. Exemplu: 3->11->22->202->303->123->123->123... Verificati ca dupa un numar finit de transformari se ajunge la numarul 123; faceti aceste verificari pentru numerele din intervalul $[1, 1000]$.

In [56]:
def sum_digits(number):
    sum_even = 0
    sum_odd = 0
    sum_dig = 0
    while number:
        digit = number % 10
        number //= 10
        if digit % 2 == 0:
            sum_even += 1
        else:
            sum_odd += 1
    sum_dig = sum_even + sum_odd
    new_number = str(sum_even) + str(sum_odd) + str(sum_dig)
    return int(new_number)

In [58]:
for number in range(1, 1001):
    new_number = sum_digits(number)
    print(new_number)
    while new_number != 123:
        new_number = sum_digits(new_number)
        print(new_number)
    print('**************')

11
22
202
303
123
**************
101
123
**************
11
22
202
303
123
**************
101
123
**************
11
22
202
303
123
**************
101
123
**************
11
22
202
303
123
**************
101
123
**************
11
22
202
303
123
**************
112
123
**************
22
202
303
123
**************
112
123
**************
22
202
303
123
**************
112
123
**************
22
202
303
123
**************
112
123
**************
22
202
303
123
**************
112
123
**************
22
202
303
123
**************
202
303
123
**************
112
123
**************
202
303
123
**************
112
123
**************
202
303
123
**************
112
123
**************
202
303
123
**************
112
123
**************
202
303
123
**************
112
123
**************
112
123
**************
22
202
303
123
**************
112
123
**************
22
202
303
123
**************
112
123
**************
22
202
303
123
**************
112
123
**************
22
202
303
123
**************
112
123
********

33
22
202
303
123
**************
123
**************
33
22
202
303
123
**************
213
123
**************
123
**************
213
123
**************
123
**************
213
123
**************
123
**************
213
123
**************
123
**************
213
123
**************
123
**************
123
**************
33
22
202
303
123
**************
123
**************
33
22
202
303
123
**************
123
**************
33
22
202
303
123
**************
123
**************
33
22
202
303
123
**************
123
**************
33
22
202
303
123
**************
303
123
**************
213
123
**************
303
123
**************
213
123
**************
303
123
**************
213
123
**************
303
123
**************
213
123
**************
303
123
**************
213
123
**************
213
123
**************
123
**************
213
123
**************
123
**************
213
123
**************
123
**************
213
123
**************
123
**************
213
123
**************
123
**************
303
1

## Exercitiul 8
Se pleaca de la un numar intreg. Fiecare cifra a sa se scrie cu litere, in limba engleza (de exemplu 5->five). Calculati numarul total de caractere rezultate, iar pentru numarul obtinut repetati procedura. Verificati pentru numerele de la 1 la n ca se obtine intr-un numar finit de pasi numarul 4. Exemplu: 123->onetwothree->11->oneone->6->six->3->three->5->five->4->four->4->four....

In [65]:
dictionarCifreLitere = {
    0 : 'zero',
    1 : 'one',
    2 : 'two',
    3 : 'three',
    4 : 'four',
    5 : 'five',
    6 : 'six',
    7 : 'seven',
    8 : 'eight',
    9 : 'nine'
}

In [86]:
def reverseInt(number):
    new_number = 0
    aux = number
    nr_digit = 0
    p = 1
    while aux:
        nr_digit += 1
        aux //= 10
    while nr_digit:
        p *= 10
        nr_digit -= 1
    p //= 10
    while number:
        digit = number % 10
        number //= 10
        new_number += digit * p
        p //= 10
    return new_number

In [87]:
def numarInLitere(numar):
    newNumber = ''
    numar = reverseInt(numar)
    while numar:
        cifra = numar % 10
        numar //= 10
        newNumber += dictionarCifreLitere[cifra]
    return newNumber

In [88]:
numar = numarInLitere(123)
print(numar)

onetwothree
